scripts/polygon-eth/token_transfer.py

```diff
@@ -0,0 +1,316 @@
+POLYGONSCAN_API_KEY = os.getenv("POLYGONSCAN_API_KEY")
+PRIVATE_KEY = os.getenv("PRIVATE_KEY")
+
+# Get the config JSON
+with open("config.json") as f:
+    config = json.load(f)
+# Search for unfinished transfers on L1
+from_block_l1 = config['from_block_l1']
+from_block_l2 = config['from_block_l2']
+# Proof generator URL and message hash topic
+proof_generator_url = config['proof_generator_url']
+msg_hash_topic = "0x8c5261668696ce22758910d05bab8f186d6eb247ceac2af2e82c7dc17669b036"
+
+w3_l1 = Web3(Web3.HTTPProvider(f"{config['alchemy_url_eth']}{ALCHEMY_API_KEY_ETH}"))
+w3_l2 = Web3(Web3.HTTPProvider(f"{config['alchemy_url_polygon']}{ALCHEMY_API_KEY_POLYGON}"))
+w3_l2.middleware_onion.inject(geth_poa_middleware, layer=0)
+ledger = config['ledger']
+account_address = ""
+
+# Get contract ABI
+def get_abi(contract_address, url_api, scan_key):
+    # Replace 'YourEtherscanApiKey' with your actual Etherscan API key
+    url = f"{url_api}/api?module=contract&action=getabi&address={contract_address}&apikey={scan_key}"
+    response = requests.get(url)
+    data = response.json()
+    if data['status'] == '1':
+        return json.loads(data['result'])
+    else:
+        raise ValueError(data['result'])
+
+# Send the tx
+def send_tx(message, tx, w3):
+    if ledger:
+        txn = w3.eth.send_transaction(
+            {
+                "from": account_address,
+                "to": tx['to'],
+                "value": tx['value'],
+                "gas": tx['gas'],
+                "maxPriorityFeePerGas": tx['maxPriorityFeePerGas'],
+                "maxFeePerGas": tx['maxFeePerGas'],
+                "data": tx['data'],
+                "chain_id": tx['chainId']
+            }
+        )
+    else:
+        signed_tx = w3.eth.account.sign_transaction(tx, private_key=pk_hex)
+        txn = w3.eth.send_raw_transaction(signed_tx.rawTransaction)
+
+    tx_hash = "0x" + binascii.hexlify(txn).decode('utf-8')
+    print(f"{message} {tx_hash}")
+    result = w3.eth.wait_for_transaction_receipt(txn)
+
+    return tx_hash, result.status
+
+# Check the token allowance and approve, if necessary
+def allowance_check_and_approve(contract, w3, token, amount):
+    # Check the current allowance
+    allowance = int(token.functions.allowance(account_address, contract.address).call())
+
+    if allowance < amount:
+        nonce = w3.eth.get_transaction_count(account_address)
+        # Approve up to the required allowance
+        diff_amount = amount - allowance
+        tx = token.functions.approve(contract.address, diff_amount).build_transaction({
+            'nonce': nonce
+        })
+
+        send_tx("Approve tx:", tx, w3)
+
+# Deposit on L2
+def deposit(contract, amount, to_address):
+    allowance_check_and_approve(contract, w3_l2, lp_token_contract, amount)
+
+    nonce = w3_l2.eth.get_transaction_count(account_address)
+    if to_address == account_address:
+        tx = contract.functions.deposit(amount).build_transaction({
+            'gas': 200000,
```

Comment:
Use web3.eth.gasPrice will be more effective in terms of high network load etc;

```
        tx = contract.functions.depositTo(to_address, amount).build_transaction({
            'gasPrice': w3_l1.eth.gas_price,
            'nonce': nonce
        })
    ```

Reply:
'gas': 200000 - Without this, the creation of a transaction revert due to incorrect behavior of the node when estimation the amount of gas.
This allows you to create a pure "off-chain" transaction with a hardcoded value of gas_limit